### DISCUSSION TITLE
Exporting Javascript Function

### DISCUSSION THREAD

**Discussion by kaffimunster** [SilentVoid13/Templater#353] [Help]
Hello!

I have been trying to follow [the documentation](https://silentvoid13.github.io/Templater/docs/user-functions/script-user-functions) to create my own script. However, even replicating the example in the documentation does not work for me. Regardless of the function export I get this error in console: `Template parsing error, aborting. tp.user.myFunction is not a function`

In addition to what is mentioned in the documentation, I have tried: 
```javascript
exports.myFunction = function(tp, tr){
    const t = tp.file.tile;
    const s = t.substring(0, 20);

    tR += s;
}

// and also 

function myFunction(tp, tR) {
    const t = tp.file.tile;
    const s = t.substring(0, 20);

    tR += s;
}
module.exports = myFunction;
```

**Comment by welpdx** [comment]
Hi,
I've noted found several issues:
1. It is `tp.file.title` not `tp.file.tile`
2. I think we can only pass `tp` to the function, we can't pass `tR`. (Idk why.)
3. In your template, call the script instead of the function. So if you have a script called `my_script.js`, in your template, do `<%* tp.user.my_script(tp);%>` instead of `<%* tp.user.myFunction(tp);%>`. (The [documentation](https://silentvoid13.github.io/Templater/docs/user-functions/script-user-functions/#functions-arguments) got it wrong. Don't put <user function> there. 

#### Try this:
In your template, do:
`<%* tR += tp.user.my_script(tp);%>`
In `my_script.js`, do:

```
function my_function(tp) {
    const t = tp.file.title;
    const s = t.substring(0, 20);
    return s;
}
module.exports = my_function;
```

Hope that helps!

~Welp
